From 857e239c3ef57e6ba0ff148de96e14249a4d669b Mon Sep 17 00:00:00 2001
Message-Id: <857e239c3ef57e6ba0ff148de96e14249a4d669b.1649937087.git.geliang.tang@suse.com>
From: Jeremy Pallotta <jmpallotta@gmail.com>
Date: Thu, 23 Dec 2021 17:23:31 -0800
Subject: [PATCH] ntb_hw_switchtec: AND with the part_map for a valid tpart_vec
Git-commit: 857e239c3ef57e6ba0ff148de96e14249a4d669b
Patch-mainline: v5.17-rc1
References: bsn#170
Modified-by-SEL: No


Some firmware versions return 1 in the target partition vector for
undefined partitions. AND with the part_map to give a valid tpart_vec.

Signed-off-by: Jeremy Pallotta <jmpallotta@gmail.com>
Signed-off-by: Kelvin Cao <kelvin.cao@microchip.com>
Signed-off-by: Jon Mason <jdmason@kudzu.us>
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 drivers/ntb/hw/mscc/ntb_hw_switchtec.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/ntb/hw/mscc/ntb_hw_switchtec.c b/drivers/ntb/hw/mscc/ntb_hw_switchtec.c
index 36efba21db4b..8e9b97631884 100644
--- a/drivers/ntb/hw/mscc/ntb_hw_switchtec.c
+++ b/drivers/ntb/hw/mscc/ntb_hw_switchtec.c
@@ -858,6 +858,7 @@ static int switchtec_ntb_init_sndev(struct switchtec_ntb *sndev)
 	tpart_vec |= ioread32(&sndev->mmio_ntb->ntp_info[self].target_part_low);
 
 	part_map = ioread64(&sndev->mmio_ntb->ep_map);
+	tpart_vec &= part_map;
 	part_map &= ~(1 << sndev->self_partition);
 
 	if (!tpart_vec) {
-- 
2.34.1

